content-api/content-actors/src/main/scala/org/sunbird/content/util/CopyManager.scala

```diff
@@ -71,21 +70,43 @@ object CopyManager {
     }
 
     def copyCollection(originNode: Node, request: Request)(implicit ec:ExecutionContext):Future[Node] = {
+        val copyType = request.getRequest.get(CopyConstants.COPY_TYPE).asInstanceOf[String]
         copyContent(originNode, request).map(node => {
             val req = new Request(request)
+            req.getContext.put("schemaName", collSchemaName)
+            req.getContext.put("version", version)
             req.put(CopyConstants.ROOT_ID, request.get(CopyConstants.IDENTIFIER))
             req.put(CopyConstants.MODE, request.get(CopyConstants.MODE))
             HierarchyManager.getHierarchy(req).map(response => {
-                val originHierarchy = response.get(CopyConstants.HIERARCHY).asInstanceOf[java.util.Map[String, AnyRef]]
-                val updateHierarchyRequest = prepareHierarchyRequest(originHierarchy, originNode, node)
-                val hierarchyRequest = new Request(request)
-                request.putAll(updateHierarchyRequest)
-                UpdateHierarchyManager.updateHierarchy(hierarchyRequest)
-                node
-            })
+                val originHierarchy = response.getResult.getOrDefault(CopyConstants.CONTENT, new java.util.HashMap[String, AnyRef]()).asInstanceOf[java.util.Map[String, AnyRef]]
+                copyType match {
+                    case CopyConstants.COPY_TYPE_SHALLOW => updateShallowHierarchy(request, node, originNode, originHierarchy)
+                    case _ => updateHierarchy(request,node, originNode, originHierarchy, copyType)
+                }
+            }).flatMap(f=>f)
         }).flatMap(f => f) recoverWith {case e: CompletionException => throw e.getCause}
     }
 
+    def updateHierarchy(request: Request, node: Node, originNode: Node, originHierarchy: util.Map[String, AnyRef], copyType:String)(implicit ec: ExecutionContext): Future[Node] = {
+        val updateHierarchyRequest = prepareHierarchyRequest(originHierarchy, originNode, node, copyType)
+        val hierarchyRequest = new Request(request)
```

Comment:
please add schema name as collection and  add version to request context

Reply:
done

Reply:
done